libp2p/muxers/mplex/lpchannel.nim

```diff
@@ -227,7 +224,11 @@ method reset*(s: LPChannel) {.base, async, gcsafe.} =
   # might be dead already - reset is always
   # optimistic
   asyncCheck s.resetMessage()
+
+  # drain the buffer before closing
+  await s.drainBuffer()
```

Comment:
so if  someone keeps adding data, this will keep draining it - is that the intent?